tap_adroll/client.py

```diff
@@ -12,27 +12,37 @@
 
 class AdrollAuthenticationError(Exception):
     pass
-
-
 class AdrollClient():
-    def __init__(self, config_path, config):
-        token = {
-            'access_token': config['access_token'],
-            'refresh_token': config['refresh_token'],
-            'token_type': 'Bearer',
-            # Set expires_in to a negative number to force the client to reauthenticate
-            'expires_in': '-30'
-        }
-        extra = {
-            'client_id': config['client_id'],
-            'client_secret': config['client_secret']
-        }
+    def __init__(self, config_path, config, dev_mode = False):
+        self.dev_mode = dev_mode
         self.config_path = config_path
-        self.session = OAuth2Session(config['client_id'],
-                                     token=token,
-                                     auto_refresh_url=TOKEN_REFRESH_URL,
-                                     auto_refresh_kwargs=extra,
-                                     token_updater=self._write_config)
+        self.config = config
+        self.authenticate_request()
+
+    def authenticate_request(self):
+        if self.dev_mode :
+            self.session = requests.Session()
```

Comment:
In the normal flow, `OAuth2Session` is used to create a new session. Should we utilize the same functionality within dev mode? In this way, we can get rid of header modification as well ([your code](https://github.com/singer-io/tap-adroll/blob/TDL-15660-implement-dev-mode/tap_adroll/client.py#L83-L85)).

Reference - [tap-quickbooks](https://github.com/singer-io/tap-quickbooks/blob/implement-dev-mode/tap_quickbooks/client.py#L164-#L165)

Reply:
made the changes